packages/anoncreds/src/services/AnonCredsIssuerService.ts

```diff
@@ -9,6 +9,8 @@ import type { AnonCredsCredentialOffer } from '../models/exchange'
 import type { AnonCredsCredentialDefinition, AnonCredsSchema } from '../models/registry'
 import type { AgentContext } from '@aries-framework/core'
 
+export const AnonCredsIssuerServiceSymbol = Symbol('AnonCredsIssuerService')
+
 export interface AnonCredsIssuerService {
   createSchema(agentContext: AgentContext, options: CreateSchemaOptions): Promise<AnonCredsSchema>
 
```

Comment:
My idea is that we add records for this in the AnonCreds package and we instruct that the same repositories are used by implementations that can use it. However, for Indy SDK we can't use the records and therefore it's part of the specific issuer/holder/verifier services.

This way there also won't be migration scripts needed, as they'll be using the same records. The idea is to create a separate record for the private parts so it's not exposed to the end user (unless they specifically fetch the private record).

See https://github.com/hyperledger/aries-framework-javascript/issues/1193 and https://github.com/hyperledger/aries-framework-javascript/issues/1164 for extra context.

Do you think this approach will work? Happy to go other directions

Reply:
Yes, I think it will be better that way, especially because we are talking about records that are aligned with AnonCreds specification, so any implementation should be fine by using it (and if it needs something else, it can probably use an internal metadata value).

The same would apply to `AnonCredsCredentialRecord`, right? I mean there will be a common Record/Repository and each `AnonCredsHolderService` implementation will use it. In such case, wouldn't be the methods `getCredential` / `deleteCredential` / `storeCredential` nearly the same for every implementation (except of indy-sdk of course)?

Reply:
Yes probably! I think `storeCredential` in indy-sdk verifies the credential so that will be different.

Reply:
```
interface AnonCredsCredentialInfo {
        credentialId: string
        attributes: Record<string, string>
        schemaId: string
        credentialDefinitionId: string
        revocationRegistryId?: string
        credentialRevocationId?: string
}

getCredential(credentialId: string): Promise<AnonCredsCredentialInfo>
deleteCredential(credentialId: string): Promise<void>
```

Reply:
```
interface AnonCredsCredentialInfo {
        credentialId: string
        attributes: Record<string, string>
        schemaId: string
        credentialDefinitionId: string
        revocationRegistryId?: string
        credentialRevocationId?: string
}

getCredential(credentialId: string): Promise<AnonCredsCredentialInfo>
deleteCredential(credentialId: string): Promise<void>
```